main.cpp

```diff
@@ -13,75 +13,584 @@ What you create in this project will be used as the basis of Project 5 in the co
+    // not sure if it's advisable to calculate within a struct, but i did have speed in MPH and KPH
+    double carTopSpeedKPH = carTopSpeedMPH * 1.609344;
+    //5) Max RPMs on engine
+    unsigned int engMaxRPM = 11300;
+    //1) start and stop engine
+    void engineStart();
+    //2) accelerate speed
+    void accelerate(unsigned int currSpeed, unsigned int topSpeed)
+    {
+        if(currSpeed < topSpeed)
+        {
+            currSpeed += 5;
+        }
+    }
+    //3) set current gear
+    int currGear();
+};
+
+/* Thing 5) Camera
+5 properties:
+    1) Focus amount
+    2) Exposure value
+    3) White balance 
+    4) Aspect ratio selection
+    5) Resolution (Width X Height)
+3 things it can do:
+    1) Enable/Disable image capture
+    2) Stream data for Tracking objects and hands
+    3) Adjust Exposure
+ */
+ struct Camera
+ {
+    //1) Focus amount
+    float focusValue = 0.5f;
+    //2) Exposure value
+    float exposureValue = 1.3f;
+    //3) White balance 
+    float whiteBalanceValue = 0.8f;
+    //4) Aspect ratio selection
+    int aspectRatioNumber = 2;
+    //5) Resolution (Width X Height)
+    struct Resolution
+    {
+        int width = 1920;
+        int height = 1080;
+    };
+    Resolution resolution;
+    //1) Enable/Disable image capture
+    void capture(bool state);
+    //2) Stream data for Tracking objects and hands
+    void captureTrackData(std::string objName, int posX, int posY);
+    //3) Adjust Exposure
+ };
+/*
+Thing 6) Board
+5 properties:
+    1) Height in cm
+    2) Width in cm
+    3) Active area (Tracked)
+    4) Inactive area (Non Tracked)
+    5) Location in virtual world space (X,Y,Z)
+3 things it can do:
+    1) Detect object on surface
+    2) Detect hand on surface
+    3) Reposition in world space
+ */
+ struct Board
+ {
+    //1) Height in cm
+    int height = 70;
+    //2) Width in cm
+    int width = 70;
+    //3) Active area (Tracked)
+    //4) Inactive area (Non Tracked) (Consolidating these two into one bool based on area)
+    bool activeArea(int horizontal, int vertical);
+    //5) Location in virtual world space (X,Y,Z)
+    struct World
+    {
+        int X;
+        int Y;
+        int Z;
+    };
+    World room;
+
+    //this throws an end of non-void function warning
+    //i would have just declared it but i wanted to
+    //do two actions in one function (hands and marker)
+    int detectObjectType(std::string object, int objHeight)
+    {
+        //1) Detect object on surface
+        if (objHeight <= 0.0f)
+        {
+            if(object == "marker")
+            {
+              return 1;
+            }
+            //2) Detect hand on surface
+            if(object == "Lhand")
+            {
+              return 2;
+            }
+            //2) Detect hand on surface
+            if(object == "Rhand")
+            {
+              return 3;
+            }
+            
+        }
+        else
+        {
+            return 0;
+        }  
```

Comment:
2 things:
1) fix the indentation of your return statements.  Code readability is very important.
2) If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
When the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise